plugins/modules/aws_acm.py

```diff
@@ -301,112 +504,68 @@ def main():
         domain_name=dict(aliases=['domain']),
         name_tag=dict(aliases=['name']),
         private_key=dict(no_log=True),
-        state=dict(default='present', choices=['present', 'absent'])
+        tags=dict(type='dict'),
+        purge_tags=dict(type='bool', default=False),
+        state=dict(default='present', choices=['present', 'absent']),
+    )
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
     )
-    required_if = [
-        ['state', 'present', ['certificate', 'name_tag', 'private_key']],
-    ]
-    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=True, required_if=required_if)
     acm = ACMServiceManager(module)
 
     # Check argument requirements
     if module.params['state'] == 'present':
-        if module.params['certificate_arn']:
-            module.fail_json(msg="Parameter 'certificate_arn' is only valid if parameter 'state' is specified as 'absent'")
+        # at least one of these should be specified.
+        absent_args = ['certificate_arn', 'domain_name', 'name_tag']
+        if sum([(module.params[a] is not None) for a in absent_args]) < 1:
+            for a in absent_args:
+                module.debug("%s is %s" % (a, module.params[a]))
+            module.fail_json(msg="If 'state' is specified as 'present' then at least one of 'name_tag', 'certificate_arn' or 'domain_name' must be specified")
     else:  # absent
         # exactly one of these should be specified
         absent_args = ['certificate_arn', 'domain_name', 'name_tag']
         if sum([(module.params[a] is not None) for a in absent_args]) != 1:
             for a in absent_args:
                 module.debug("%s is %s" % (a, module.params[a]))
-            module.fail_json(msg="If 'state' is specified as 'absent' then exactly one of 'name_tag', certificate_arn' or 'domain_name' must be specified")
-
-    if module.params['name_tag']:
-        tags = dict(Name=module.params['name_tag'])
-    else:
-        tags = None
+            module.fail_json(msg="If 'state' is specified as 'absent' then exactly one of 'name_tag', 'certificate_arn' or 'domain_name' must be specified")
+
+    filter_tags = None
+    desired_tags = None
+    if module.params.get('tags') is not None:
+        desired_tags = module.params['tags']
+    if module.params.get('name_tag') is not None:
+        # The module was originally implemented to filter certificates based on the 'Name' tag.
+        # Other tags are not used to filter certificates.
+        # It would make sense to replace the existing name_tag, domain, certificate_arn attributes
+        # with a 'filter' attribute, but that would break backwards-compatibility.
```

Comment:
@markuman @alinabuzachis what do you think about adding a deprecation notice for the attributes mentioned here ? Maybe one for a separate PR.

Reply:
Can this be handled separately?

Reply:
@marknet15 I guess we should add a deprecation notice. I am still confused because `signature_algorithms` doesn't seem to be documented. So, I guess we need to update the documentation with this param and the new one that replaces it and add a deprecation notice for `signature_algorithms`. Not sure if this makes sense @markuman .

Reply:
What else do I need to do? This is stalling.

Reply:
@markuman how would you suggest to proceed here?

Reply:
Can this conversation be resolved? I don't see any change being requested for this PR.